### PR TITLE
Conda-build 2.0 update

### PR DESCRIPTION
**Pull request by msarahan**
This is the first of probably a few PRs for preparing for conda-build 2.0 release.  I wouldn't bother reviewing this yet.

### CODE REVIEW

**Comment by msarahan** [comment]
OK @electronwill @kerrywatson1 - this is ready for some review.  I will have another PR sometime in the near future to add details on the new conda-build API.

**Comment by kerrywatson1** [comment]
The discussion on 'older syntax' versus 'newer syntax' is not clear. If you can, call out specific versions that use one vs. the other. That would take extra digging and testing.

File: docs/source/building/meta-yaml.rst [hunk 1]
```diff
@@ -110,6 +110,14 @@ will be copied to the work directory before building:
   source:
     path: ../src
 
+If the local path is a git or svn repository, you will get the corresponding environment
+variables defined in your build environment.  The only practical difference between
+``git_url`` or ``hg_url`` and ``path`` as source arguments is that ``git_url`` and ``hg_url``
+would be clones of a repository and path is copy of the repository.  Using path will allow you
```
Comment (electronwill) [inline docs/source/building/meta-yaml.rst]: and path is copy -> and ``path`` is a copy

File: docs/source/building/meta-yaml.rst [hunk 2]
```diff
@@ -110,6 +110,14 @@ will be copied to the work directory before building:
...
+to build packages with unstaged/uncommited changes in working directory. git_url can only
```
Comment (electronwill) [inline docs/source/building/meta-yaml.rst]: uncommited -> uncommitted
in working directory. -> in the working directory.

**Comment by electronwill** [comment]
I made a couple of small suggestions, but this looks very good.

If I understand correctly, the older syntax required users to type meta.yaml lines such as "version: {{ environ.get('GIT_DESCRIBE_TAG', '') }}", and now they can either continue doing that, or instead use the newer syntax and just type "version: {{ GIT_DESCRIBE_TAG }}". Both will produce the same effect, although the edited docs here do say "the newer syntax checks to make sure the variable is defined and produces a clear error if it is not."

**Comment by electronwill** [comment]
I just noticed that all the small suggestions are on text that you didn't write, and were just moving around. Not your fault, but we might as well fix them while we're moving them.

**Comment by msarahan** [comment]
I think I'm going to rewrite that bit.  It's much less about old and new,
and more simply about "do you want to set a default value, or not?"

**Comment by electronwill** [comment]
That makes sense. I did notice that the newer way didn't include the 0 or ''.

File: docs/source/building/meta-yaml.rst [hunk 3]
```diff
@@ -603,10 +603,12 @@ at the base directory of the git repository, so the ``git_url`` is ``../``:
      source:
        git_url: ../
 
-It is also possible to use the older syntax for these environment variables,
-although it is somewhat more verbose, and the newer syntax checks to make sure
-the variable is defined and produces a clear error if it is not. This is the
-example above using the older syntax:
+Conda-build will make sure the jinja2 variables you use are defined and produces
```
Comment (electronwill) [inline docs/source/building/meta-yaml.rst]: produces -> produce

File: docs/source/building/meta-yaml.rst [hunk 4]
```diff
@@ -603,10 +603,12 @@ at the base directory of the git repository, so the ``git_url`` is ``../``:
...
+a clear error if it is not.
+
+It is also possible to use a different syntax for these environment variables
+that allows default values to be set, although it is somewhat more verbose.  This is the
+example above using the other syntax:
```
Comment (electronwill) [inline docs/source/building/meta-yaml.rst]: the other syntax -> the syntax that allows defaults

**Comment by electronwill** [comment]
Good catch on adding quotes to "git_url".

**Comment by electronwill** [comment]
This looks good to me!

**Comment by msarahan** [comment]
Thanks @electronwill - when will this go live?

**Comment by kerrywatson1** [comment]
Mike, docs go live daily, and occasionally more often. So these will be
live in a few hours.


_______________________________

Kerry Watson
Documentation manager

[image: Continuum Analytics] <http://continuum.io>

**Comment by msarahan** [comment]
Thanks, Kerry.  I'm asking because I'm updating internal conda-build links
to point to the new content.  Later today will be just fine.

**Comment by electronwill** [comment]
Thank you @msarahan ! Most ContinuumIO product docs have to be deployed by hand, but the conda-docs repo is set up with CI deployment already, so as soon as the PR is merged the CI job starts, and as soon as the CI job finishes the changes go live. I just checked the site and these changes are live now.

**Pull request merged**